packages/repo-tools/src/commands/api-reports/sql-reports/runSqlExtraction.ts

```diff
@@ -0,0 +1,212 @@
+/*
+ * Copyright 2024 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import fs, { readJson } from 'fs-extra';
+import { relative as relativePath } from 'path';
+import { paths as cliPaths } from '../../../lib/paths';
+import { diff as justDiff } from 'just-diff';
+import { SchemaInfo } from './types';
+import { getPgSchemaInfo } from './getPgSchemaInfo';
+import { generateSqlReport } from './generateSqlReport';
+import type { Knex } from 'knex';
+import { logApiReportInstructions } from '../api-extractor';
+
+interface SqlExtractionOptions {
+  packageDirs: string[];
+  isLocalBuild: boolean;
+}
+
+export async function runSqlExtraction(options: SqlExtractionOptions) {
+  const { default: Knex } = await import('knex');
+  const { default: ClientPgLite } = await import('knex-pglite');
+
+  // Since we're passing this as the client we need to replace the `config.client` with `pg` afterwards
+  class WrappedClientPgLite extends ClientPgLite {
+    constructor(config: any) {
+      super({ ...config, client: 'pg' });
+    }
+  }
+
+  let dbIndex = 1;
+
+  for (const packageDir of options.packageDirs) {
+    const migrationDir = cliPaths.resolveTargetRoot(packageDir, 'migrations');
+    if (!(await fs.pathExists(migrationDir))) {
+      console.log(`No SQL migrations found in ${packageDir}`);
+      continue;
+    }
+
+    const { name: pkgName } = await readJson(
+      cliPaths.resolveTargetRoot(packageDir, 'package.json'),
+    );
+
+    const migrationFiles = await fs.readdir(migrationDir, {
+      withFileTypes: true,
+    });
+
+    const migrationTargets = migrationFiles
+      .filter(entry => entry.isDirectory())
+      .map(entry => entry.name);
+    if (migrationFiles.some(entry => entry.isFile())) {
+      migrationTargets.push('.');
+    }
+
+    for (const migrationTarget of migrationTargets) {
+      const database = `extractor-${dbIndex++}`;
+
+      const knex = Knex({
+        client: WrappedClientPgLite,
+        dialect: 'postgres',
+        connection: {
+          database,
+        },
+      });
+
+      await knex.raw(`CREATE DATABASE "${database}"`);
```

Comment:
```suggestion
      await knex.raw('CREATE DATABASE ??', [database]);
```

not that it matters much